src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2076,6 +2076,16 @@ protected override void HandleDisassembleRequestAsync(IRequestResponder<Disassem
                         Instruction = data.bstrOpcode,
                         Symbol = data.bstrSymbol
                     };
+
+                    if (!string.IsNullOrWhiteSpace(data.bstrDocumentUrl))
+                    {
```

Comment:
This should be checking for the hasSource flag, and it should either return `data.bstrDocumentUrl` (if set) or the last `bstrDocumentUrl` if the current one is null.

If you want to reduce the data payload a bit, you could also omit this if the line and document didn't change.

Reply:
Wait, that wasn't quite right. We can set `instruction.Location` whenever `data.bstrDocumentUrl` is not null or empty. It is just the line/endline etc that should be based on `DF_HASSOURCE`.